mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -2054,26 +2083,60 @@ impl MakerOrder {
                 let ticker_match = (self.base == taker.base
                     || self.base_orderbook_ticker.as_ref() == Some(&taker.base))
                     && (self.rel == taker.rel || self.rel_orderbook_ticker.as_ref() == Some(&taker.rel));
+                // taker_base_amount: the amount taker desires to buy (input.volume from SellBuyRequest)
+                // taker_rel_amount: the amount  taker is willing to pay (input.volume * input.price, where input is SellBuyRequest)
+                // taker_price: the effective price offered by the taker
                 let taker_price = taker_rel_amount / taker_base_amount;
-                if ticker_match
-                    && taker_base_amount <= &self.available_amount()
-                    && taker_base_amount >= &self.min_base_vol
-                    && taker_price >= self.price
-                {
+
+                // Reject if any basic conditions are not satisfied
+                let base_amount_exceeds = taker_base_amount > &self.available_amount();
+                let below_min_volume = taker_base_amount < &self.min_base_vol;
+                let price_too_low = taker_price < self.price;
+
+                if !ticker_match || base_amount_exceeds || below_min_volume || price_too_low {
+                    return OrderMatchResult::NotMatched;
+                }
+
+                if self.swap_version.is_legacy() || taker.swap_version.is_legacy() {
```

Comment:
make this a var and reuse, e.g 
```rust
let is_legacy = self.swap_version.isi_legacy() || taker.swap_version.is_legacy();
```

Reply:
Done https://github.com/KomodoPlatform/komodo-defi-framework/pull/2405/commits/011798aca13b0cbe582a08fd6f1509f4c08f6b44